Tasks/VsTest/versionfinder.ts

```diff
@@ -128,46 +127,30 @@ function getLatestVSTestConsolePathFromRegistry(): Promise<string> {
             let selectedVersion = versions[versions.length - 1];
             tl.debug('Registry entry found. Selected version is ' + selectedVersion.toString());
             return Promise.resolve(getVSTestLocation(selectedVersion));
-        } 
+        }
 
         tl.debug('No Registry entry found under VisualStudio node');
         return Promise.resolve(null);
     }).on('error', () => {
         tl.debug('Registry entry not found under VisualStudio node');
         return Promise.resolve(null);
     });
-    
     return Promise.resolve(null);
 }
 
-function getVSTestConsole15Path(vs15HelperPath: string): string {
-    let powershellTool = tl.tool('powershell');
-    let powershellArgs = ['-NonInteractive', '-ExecutionPolicy', 'Unrestricted', '-file', vs15HelperPath]
-    powershellTool.arg(powershellArgs);
-    let xml = powershellTool.execSync().stdout;
-    let deferred = Q.defer<string>();
-    let vstestconsolePath: string = null;
-    xml2js.parseString(xml, (err, result) => {
-        if (result) {
-            try {
-                let vs15InstallDir = result['Objs']['S'][0];
-                vstestconsolePath = path.join(vs15InstallDir, 'Common7', 'IDE', 'CommonExtensions', 'Microsoft', 'TestWindow');
-            } catch (e) {
-                tl.debug('Unable to read Visual Studio 2017 installation path');
-                tl.debug(e);
-                vstestconsolePath = null;
-            }
-        }
-    });
-
-    return vstestconsolePath;
+function getVSTestConsole15Path(): string {
+    const vswhereTool = tl.tool(path.join(__dirname, 'vswhere.exe'));
+    vswhereTool.line('-latest -products * -requires Microsoft.VisualStudio.PackageGroup.TestTools.Core -property installationPath');
+    const op = vswhereTool.execSync().stdout;
+    tl.debug('Visual Studio 15.0 or higher installed path: ' + op);
+    return op;
```

Comment:
@ericsciple  by default it looks for 15.0 + versions only. If it needed to located 15.0 or below versions we should pass `-legacy` as param.
Correct me If I am wrong -> Going to commit into master :)